core/shared/src/main/scala-3/zio/prelude/Liftables.scala

```diff
@@ -28,27 +28,36 @@ trait Liftables {
-        case '{ Assertion.Regex.literal(${Expr(str)}) }                                    => Some(Assertion.Regex.literal(str))
-        case '{ Assertion.Regex.anyOf(${Expr(first)}, ${Expr(second)}, ${Expr(rest)}) }    => Some(Assertion.Regex.anyOf(first, second, rest))
-        case '{ Assertion.Regex.notAnyOf(${Expr(first)}, ${Expr(second)}, ${Expr(rest)}) } => Some(Assertion.Regex.notAnyOf(first, second, rest))
-        case '{ Assertion.Regex.inRange(${Expr(start)}, ${Expr(end)}) }                    => Some(Assertion.Regex.inRange(start, end))
-        case '{ Assertion.Regex.notInRange(${Expr(start)}, ${Expr(end)}) }                 => Some(Assertion.Regex.notInRange(start, end))
-        case '{ (${Expr(regex)}: Assertion.Regex).+ }                                      => Some(regex.+)
-        case '{ (${Expr(regex)}: Assertion.Regex).* }                                      => Some(regex.*)
-        case '{ (${Expr(regex)}: Assertion.Regex).between(${Expr(min)}, ${Expr(max)}) }    => Some(regex.between(min, max))
-        case '{ (${Expr(regex)}: Assertion.Regex).min(${Expr(n)}) }                        => Some(regex.min(n))
-        case '{ (${Expr(regex)}: Assertion.Regex).max(${Expr(n)}) }                        => Some(regex.max(n))
-        case '{ (${Expr(left)}: Assertion.Regex).~(${Expr(right)}: Assertion.Regex) }     => Some(left ~ right)
-        case '{ (${Expr(left)}: Assertion.Regex).|(${Expr(right)}: Assertion.Regex) }     => Some(left | right)
-        case _                                                                              => None
+        case '{ Assertion.Regex.anyChar }                                                       => Some(Assertion.Regex.anyChar)
+        case '{ Assertion.Regex.anything }                                                      => Some(Assertion.Regex.anything)
+        case '{ Assertion.Regex.alphanumeric }                                                  => Some(Assertion.Regex.alphanumeric)
+        case '{ Assertion.Regex.nonAlphanumeric }                                               => Some(Assertion.Regex.nonAlphanumeric)
+        case '{ Assertion.Regex.whitespace }                                                    => Some(Assertion.Regex.whitespace)
+        case '{ Assertion.Regex.nonWhitespace }                                                 => Some(Assertion.Regex.nonWhitespace)
+        case '{ Assertion.Regex.digit }                                                         => Some(Assertion.Regex.digit)
+        case '{ Assertion.Regex.nonDigit }                                                      => Some(Assertion.Regex.nonDigit)
+        case '{ Assertion.Regex.literal(${Expr(str)}) }                                         => Some(Assertion.Regex.literal(str))
+        case '{ Assertion.Regex.anyCharOf(${Expr(first)}, ${Expr(second)}) }                    => Some(Assertion.Regex.anyCharOf(first, second))
```

Comment:
Added this case to solve macro expansion errors when just passing two arguments to `anyCharOf`

Reply:
I believe there's a way to match on and unpack `VarArgs`: Check out the **Working with Varargs** section in https://docs.scala-lang.org/scala3/guides/macros/macros.html

Reply:
Ahhh yes! That's right